### PR TITLE
Add masked_fill_.Scalar, masked_fill_.Tensor

### PR DESCRIPTION
**Pull request by wconstab**
- Fixed codegenerator issue where IR class would not be generated for
  masked_fill_.Tensor overload

#65576

### CODE REVIEW

File: lazy_tensor_core/lazy_tensor_core/csrc/LazyShapeDtype.cpp [hunk 1]
```diff
@@ -51,6 +51,18 @@ namespace torch_lazy_tensors {
 namespace ir {
 namespace ops {
 
+std::vector<std::vector<int64_t>> compute_shape_masked_fill(at::Tensor & self, const at::Tensor & mask, const at::Scalar & value) {
+  return {self.sizes().vec()};
+}
+std::vector<c10::ScalarType> compute_dtype_masked_fill(at::Tensor & self, const at::Tensor & mask, const at::Scalar & value) {
+  return {value.type()};
```
Comment (alanwaketan) [inline lazy_tensor_core/lazy_tensor_core/csrc/LazyShapeDtype.cpp:58]: Could the dtype of self different than the dtype of value?

File: tools/codegen/gen_lazy_tensor.py [hunk 2]
```diff
@@ -110,6 +110,11 @@ def concat_map_codegen(func: Callable[[NativeFunction], Sequence[str]],
         TODO(alanwaketan): Once all ops are grouped properly, we should no longer need this hack.
         """
         generated = set()
+        def gen_key(func: NativeFunction):
+            # we want to generate unique entries for overloads of functional variants,
+            # but not for inplace variants unless explicitly told `codegenInplaceVariant`
+            return (func.name.name.base, func.name.overload_name)
```
Comment (alanwaketan) [inline tools/codegen/gen_lazy_tensor.py:116]: Good catch!

**Review by alanwaketan — APPROVED** [review]
Overall, LGTM. Please fix the linter issue as well.

File: lazy_tensor_core/lazy_tensor_core/csrc/LazyShapeDtype.cpp (hunk 1 above)
Comment (wconstab) [reply lazy_tensor_core/lazy_tensor_core/csrc/LazyShapeDtype.cpp:58]: I guess so. let me double check my assumption. i assumed that value's dtype would overwrite whatever self's is.

Comment (wconstab) [reply lazy_tensor_core/lazy_tensor_core/csrc/LazyShapeDtype.cpp]: oops, good catch. i was way off. It is always self.type, and if new value does not fit, it throws an error

**Pull request merged**